File: LPS.java
class LPS {
    public String longestPalindrome(String s) {
        if (s.length() == 0 || s == null) {
            return s;
        }

        String longest = "";
        for (int i = 0; i < s.length(); i++) {
            String odd = expandFromCenter(s, i, i + 1);
            String even = expandFromCenter(s, i, i);

            if (odd.length() > longest.length()) {
                longest = odd;
            } else {
                longest = even;
            }
        }
        return longest;
    }

    public String expandFromCenter(String s, int left, int right) {
        while (left >= 0 && right < s.length() && s.charAt(left) == s.charAt(right)) {
            left--;
            right++;
        }
        return s.substring(left + 1, right);
    }
}

File: LPS.py
class Solution:
    def longest(self,s):
        res=""
        for i in xrange(len(s)):
            odd = self.helper(s,i,i)
            even = self.helper(s,i,i+1)

            if len(odd) > len(res):
                res = odd
            if len(even) > len(res):
                res = even

        return res

    def helper(self,s,i,j):
        while i>=0 and r<len(s) and s[i] == s[r]:
            i-=1
            j+=1
        return s[i+1:j]